src/c++/perf_analyzer/test_perf_utils.cc

```diff
@@ -114,4 +119,254 @@ TEST_CASE("testing the ParseTensorFormat function")
+        std::make_pair("DT_BFLOAT16", "BF16"),
+        std::make_pair("DT_FLOAT", "FP32"),
+        std::make_pair("DT_DOUBLE", "FP64"),
+        std::make_pair("DT_INT32", "INT32"),
+        std::make_pair("DT_INT16", "INT16"),
+        std::make_pair("DT_INT8", "INT8"),
+        std::make_pair("DT_UINT8", "UINT8"),
+        std::make_pair("DT_STRING", "BYTES"),
+        std::make_pair("DT_INT64", "INT64"),
+        std::make_pair("DT_BOOL", "BOOL"),
+        std::make_pair("DT_UINT32", "UINT32"),
+        std::make_pair("DT_UINT64", "UINT64")};
+
+    for (const auto& type_pair : tf_to_datatype) {
+      status = ConvertDTypeFromTFS(type_pair.first, &datatype);
+      CHECK(status.IsOk());
+      CHECK(datatype == type_pair.second);
+    }
+  }
+
+  SUBCASE("Invalid tensorflow datatype")
+  {
+    status = ConvertDTypeFromTFS("dt_bool", &datatype);
+    CHECK(!status.IsOk());
+    CHECK(datatype == "");
+
+    status = ConvertDTypeFromTFS("dt_uint8", &datatype);
+    CHECK(!status.IsOk());
+    CHECK(datatype == "");
+
+    status = ConvertDTypeFromTFS("abcdef", &datatype);
+    CHECK(!status.IsOk());
+    CHECK(datatype == "");
+
+    status = ConvertDTypeFromTFS("", &datatype);
+    CHECK(!status.IsOk());
+    CHECK(datatype == "");
+  }
+}
+
+TEST_CASE("perf_utils: IsDirectory")
+{
+  // Create a temporary directory /tmp/abcdef1234
+  int status;
+  std::string temp_path{"/tmp/abcdef1234"};
+
+  CHECK(!IsDirectory(temp_path));
+
+  status = mkdir(temp_path.c_str(), S_IRWXU | S_IROTH | S_IXOTH);
+  REQUIRE(status == 0);
+  CHECK(IsDirectory(temp_path));
+
+  status = rmdir(temp_path.c_str());
+  REQUIRE(status == 0);
+  CHECK(!IsDirectory(temp_path));
+}
+
+TEST_CASE("perf_utils: IsFile")
+{
+  // Create a temporary file /tmp/test.txt
+  int status;
+  std::string temp_path{"/tmp/test.txt"};
+
+  CHECK(!IsFile(temp_path));
+
+  std::ofstream file(temp_path);
+  CHECK(IsFile(temp_path));
+
+  std::remove(temp_path.c_str());
+  CHECK(!IsFile(temp_path));
+}
+
+TEST_CASE("perf_utils: ByteSize")
+{
+  std::vector<int64_t> shape{3, 4, 5};
+
+  SUBCASE("Single byte elements")
+  {
+    CHECK(ByteSize(shape, "BOOL") == 1 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "INT8") == 1 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT8") == 1 * 3 * 4 * 5);
+  }
+
+  SUBCASE("2 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "BF16") == 2 * 3 * 4 * 5);
+  }
+
+  SUBCASE("4 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT32") == 4 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT32") == 4 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP32") == 4 * 3 * 4 * 5);
+  }
+
+  SUBCASE("8 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT64") == 8 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT64") == 8 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP64") == 8 * 3 * 4 * 5);
+  }
```

Comment:
How about either compute or define product once and re-use it instead of all the magic numbers?
```cpp
  std::vector<int64_t> shape{3, 4, 5};
  int num_elements = std::accumulate(shape.begin(), shape.end(), 1, std::multiplies<int>());
  // or 
  // constexpr int num_elements = 3 * 4 * 5;

  SUBCASE("Single byte elements")
  {
    CHECK(ByteSize(shape, "BOOL") == 1 * num_elements);
    CHECK(ByteSize(shape, "INT8") == 1 * num_elements);
    CHECK(ByteSize(shape, "UINT8") == 1 * num_elements);
  }

  SUBCASE("2 byte elements")
  {
    CHECK(ByteSize(shape, "INT16") == 2 * num_elements);
    CHECK(ByteSize(shape, "UINT16") == 2 * num_elements);
    CHECK(ByteSize(shape, "FP16") == 2 * num_elements);
    CHECK(ByteSize(shape, "BF16") == 2 * num_elements);
  }

  ...
```

Reply:
I agree. Removing magic numbers sounds like a good idea.